packages/block-editor/src/store/private-selectors.js

```diff
@@ -477,16 +478,37 @@ export const getContentLockingParent = createSelector(
 		let current = clientId;
 		let result;
 		while ( ( current = state.blocks.parents.get( current ) ) ) {
-			if (
-				getBlockName( state, current ) === 'core/block' ||
-				getTemplateLock( state, current ) === 'contentOnly'
-			) {
+			if ( isContentLockingParent( state, current ) ) {
 				result = current;
 			}
 		}
 		return result;
 	},
-	( state ) => [ state.blocks.parents, state.blockListSettings ]
+	( state ) => [
+		state.blocks.parents,
+		state.blockListSettings,
+		state.editorMode,
+		getSectionRootClientId( state ),
+	]
+);
+
+export const isContentLockingParent = createSelector(
```

Comment:
Question: Does memoizing a boolean value selector have any meaningful impact on performance?

Reply:
You're right, no impact :)